specs/trip_dispatch_spec.rb

```diff
@@ -88,5 +89,40 @@
       passenger.must_be_instance_of RideShare::Passenger
       passenger.trips.must_include trip
     end
-  end
-end
+  end# loader method
+
+  describe 'request_trip' do
+    it 'locates a driver whose status is AVAILABLE' do
+      # Arrange
+      dispatcher = RideShare::TripDispatcher.new
+      passenger_id = 1
+
+      data = {
+        start_time: Time.now,
+        rating: nil,
+        end_time: nil
+      }
+
+      # Act
+      new_request = dispatcher.request_trip(passenger_id)
+      # Assert
+      new_request.must_be_instance_of RideShare::Trip
+      new_request.driver.status.must_equal :UNAVAILABLE
+      new_request.driver.trips.must_include new_request
+      new_request.passenger.trips.must_include new_request
+
+    end # driver AVAILABLE
+
+    it "raises an ArgumentError if passenger_id is not found" do
+      dispatcher = RideShare::TripDispatcher.new
+      passenger_id = dispatcher.passengers.last.id + 1
```

Comment:
There are many more things you should be testing for the `request_trip` method:

- Does the trip reference the correct passenger and driver?
- Was the trip added to the trip lists of the driver and passenger?
- What if there is no available driver?

The last is an important edge case, and is exactly the sort of situation where software could do something unexpected. You might set up the test as follows:

```ruby
it 'raises an error if no driver is available' do
  dispatcher = RideShare::TripDispatcher.new
  dispatcher.drivers.each do |driver|
    # This may require an attr_writer in Driver
    driver.status = :UNAVAILABLE
  end
  proc {
    dispatcher.request_trip(2)
  }.must_raise StandardError
end
```